File: backend/app/schemas/order.py
"""
订单相关Schema
"""

from datetime import datetime
from typing import Optional, Dict, Any
from pydantic import BaseModel, Field


class OrderCreate(BaseModel):
    """创建订单"""
    commodity_id: int = Field(..., description="商品ID")
    quantity: int = Field(1, ge=1, description="购买数量")
    payment_id: int = Field(..., description="支付方式ID")
    contact: str = Field(..., min_length=1, description="联系方式")
    password: Optional[str] = Field(None, description="查单密码")
    race: Optional[str] = Field(None, description="商品种类")
    card_id: Optional[int] = Field(None, description="预选卡密ID")
    coupon: Optional[str] = Field(None, description="优惠券码")
    widget: Optional[Dict[str, Any]] = Field(None, description="自定义控件数据")


class OrderResponse(BaseModel):
    """订单创建响应"""
    trade_no: str
    amount: float
    status: int
    payment_url: Optional[str] = None
    payment_type: Optional[str] = None
    extra: Optional[Dict[str, Any]] = None
    secret: Optional[str] = None


class OrderDetailResponse(BaseModel):
    """订单详情响应"""
    id: int
    trade_no: str
    amount: float
    quantity: int
    status: int
    delivery_status: int
    contact: str
    secret: Optional[str] = None
    commodity_name: Optional[str] = None
    payment_name: Optional[str] = None
    created_at: Optional[datetime] = None
    paid_at: Optional[datetime] = None
    has_password: bool = False
    
    class Config:
        from_attributes = True


class OrderQuery(BaseModel):
    """订单查询"""
    contact: str = Field(..., description="联系方式或订单号")


class GetSecretRequest(BaseModel):
    """获取卡密"""
    password: Optional[str] = Field(None, description="查单密码")


class DeliverOrderRequest(BaseModel):
    """手动发货"""
    secret: str = Field(..., description="发货内容")

File: frontend/src/api/order.ts
import api from './index'

export interface CreateOrderRequest {
  commodity_id: number
  quantity: number
  payment_id: number
  contact: string
  password?: string
  race?: string
  card_id?: number
  coupon?: string
  widget?: Record<string, any>
}

export interface CreateOrderResponse {
  trade_no: string
  amount: number
  status: number
  payment_url?: string
  payment_type?: string  // redirect | qrcode | form
  extra?: Record<string, any>  // form_data, qrcode_url, etc.
  secret?: string  // 余额支付成功时返回卡密
}

export interface OrderDetail {
  id: number
  trade_no: string
  amount: number
  quantity: number
  status: number
  delivery_status: number
  contact: string
  secret?: string
  created_at: string
  paid_at?: string
  commodity_name?: string
  payment_name?: string
  has_password?: boolean
}

// 创建订单
export const createOrder = (data: CreateOrderRequest): Promise<CreateOrderResponse> => {
  return api.post('/orders/create', data)
}

// 查询订单
export const getOrder = (tradeNo: string): Promise<OrderDetail> => {
  return api.get(`/orders/${tradeNo}`)
}

// 获取卡密
export const getOrderSecret = (tradeNo: string, password?: string): Promise<{ secret: string }> => {
  return api.post(`/orders/${tradeNo}/secret`, { password })
}

// 按联系方式查询订单
export const queryOrders = (contact: string, page = 1, limit = 10): Promise<{ items: OrderDetail[] }> => {
  return api.post('/orders/query', { contact }, { params: { page, limit } })
}
